pyhap/accessory.py

```diff
@@ -193,15 +196,29 @@ def _set_services(self):
             always call the base implementation first, as it reserves IID of
             1 for the Accessory Information service (HAP requirement).
         """
+        info_data = {
```

Comment:
Shouldn't this be the default result of `get_accessory_information`? Otherwise, if you call that method, you will not get the actual accessory information in the default (not override) case.

Reply:
I was trying to get away from having to call `super()` in that method. Maybe it should be something about custom accessory information.

Reply:
I also wonder if we could get rid of the `super()` requirement from `_set_services()`, or look at some other way of defining an extra service.

Perhaps an `@add_service()` decorator, which sets up a service and also acts as a property - so we could have nice definitions. Just thought of this now, will write up an example later.

Reply:
Why are you trying to get away from the `super`?

Reply:
Mainly because it's easy to accidentally forget to call `super()` (or not call it until after other stuff has happened), whereas this code _must_ run (ie, AccessoryInformation _must_ be IID=1).

I think it would be better to have a hook that you can just override, or some other way of registering services. I'll think more about that.

Reply:
I also think that a simpler function that returns just the things that we want to override would make implementing that method (in the subclass) simpler.

Reply:
I see your point, I will try to address this.

The IID=1 requirement is a big hack as is currently implemented. I will surely kick this out from here and make something that is harder to screw up.